docs/book/src/blockchain-development/native_assets.md

```diff
@@ -1,32 +1,457 @@
+
+The asset id of any Native Asset on Fuel is calculated by taking the SHA256 hash digest of the originating `ContractId` that minted the asset and a `SubId` i.e. `sha256((contract_id, sub_id))`.
+
+### Creating a New `AssetId`
+
+There are 3 ways to instantiate a new `AssetId`:
+
+#### Default
+
+When a contract will only ever mint a single asset, it is recommended to use the `DEFAULT_ASSET_ID` sub id. This is referred to as the default asset of a contract.
+
+To get the default asset from an internal contract call, call the `default()` function:
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:default_asset_id}}
+```
+
+#### New
+
+If a contract mints multiple assets or if the asset has been minted by an external contract, the `new()` function will be needed. The `new()` function takes the `ContractId` of the contract which minted the token as well as a `SubId`.
+
+To create a new `AssetId` using a `ContractId` and `SubId`, call the `new()` function:
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:new_asset_id}}
+```
+
+#### From
+
+In the case where the `b256` value of an asset is already known, you may call the `from()` function with the `b256` value.
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:from_asset_id}}
+```
+
+## The `SubId` type
+
+The SubId is used to differentiate between different assets that are created by the same contract. The `SubId` is a `b256` value.
+
+When creating an single new asset on Fuel, we recommend using the `DEFAULT_ASSET_ID`.
+
+## The Base Asset
+
+On the Fuel Network, the base asset is Ether. This is the only asset on the Fuel Network that does not have a `SubId`.
+
+The Base Asset can be returned anytime by calling the `base_asset_id()` function of the `AssetId` type.
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:base_asset}}
+```
+
+## Basic Native Asset Functionality
+
+### Minting A Native Asset
+
+To mint a new asset, the `std::asset::mint()` function must be called internally within a contract. A `SubId` and amount of coins must be provided. These newly minted coins will be owned by the contract which minted them. To mint another asset from the same contract, replace the `DEFAULT_SUB_ID` with your desired `SubId`.
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:mint_asset}}
+```
+
+You may also mint an asset to a specific entity with the `std::asset::mint_to()` function. Be sure to provide a target `Identity` that will own the newly minted coins.
+
+```sway
+{{#include ../../../../examples/native_asset/src/main.sw:mint_to_asset}}
+```
+
+### Burning a Native Asset
+
+To burn an asset, the `std::asset::burn()` function must be called internally from the contract which minted them. The `SubId` used to mint the coins and amount must be provided. The burned coins must be owned by the contract.
```

Comment:
It would be nice to clarify what exactly happens in the background when this function gets called.

Reply:
Could you give an example? Not sure what I could include on the background functionality as it's just calling the `burn` opcode in assembly

Reply:
Just a simple explanation for what it means to burn something for a reader that maybe is new to blockchain.

Reply:
Added.